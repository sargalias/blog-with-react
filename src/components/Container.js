import React from 'react';

const Container = (props) => (
  props.children && (
    <div className="container">
      { props.children }
    </div>
  )
);

export default Container;
