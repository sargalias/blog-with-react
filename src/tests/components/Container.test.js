import React from 'react';
import Container from '../../components/Container';
import { shallow } from 'enzyme';

test('Container should be empty with no children', () => {
  const wrapper = shallow(<Container />);
  expect(wrapper.html()).toBeNull();
});

test('Container should contain contain children JSX', () => {
  const children = <div>Test HTML</div>;
  const wrapper = shallow(<Container>{ children }</Container>);
  expect(wrapper.contains(children)).toBe(true);
});

test('Container should render empty with no children', () => {
  const wrapper = shallow(<Container />);
  expect(wrapper).toMatchSnapshot();
});

test('Container should render correctly with children', () => {
  const children = <div>Test HTML</div>;
  const wrapper = shallow(<Container>{ children }</Container>);
  expect(wrapper).toMatchSnapshot();
});
