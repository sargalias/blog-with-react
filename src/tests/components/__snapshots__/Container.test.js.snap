// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render correctly with children 1`] = `
<div
  className="container"
>
  <div>
    Test HTML
  </div>
</div>
`;

exports[`Container should render empty with no children 1`] = `""`;
